x/storage/keeper/keeper.go

```diff
@@ -892,19 +966,24 @@ func (k Keeper) DeleteObject(
 		}
 		return types.ErrObjectNotSealed
 	}
-
 	// check permission
 	effect := k.VerifyObjectPermission(ctx, bucketInfo, objectInfo, operator, permtypes.ACTION_DELETE_OBJECT)
 	if effect != permtypes.EFFECT_ALLOW {
 		return types.ErrAccessDenied.Wrapf(
 			"The operator(%s) has no DeleteObject permission of the bucket(%s), object(%s)",
 			operator.String(), bucketName, objectName)
 	}
-
-	spInState := k.MustGetPrimarySPForBucket(ctx, bucketInfo)
+	if objectInfo.IsUpdating {
+		shadowObjectInfo := k.MustGetShadowObjectInfo(ctx, bucketInfo.BucketName, objectInfo.ObjectName)
+		err := k.UnlockShadowObjectFeeAndDeleteShadowObjectInfo(ctx, bucketInfo, shadowObjectInfo, objectInfo.ObjectName)
+		if err != nil {
+			return err
+		}
+	}
+	_ = k.MustGetPrimarySPForBucket(ctx, bucketInfo)
```

Comment:
This line `_ = k.MustGetPrimarySPForBucket(ctx, bucketInfo)` is not needed.

Reply:
removed